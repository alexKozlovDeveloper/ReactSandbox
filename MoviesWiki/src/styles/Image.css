.image {
    width: 100%;
}

.imagecontainer {

}