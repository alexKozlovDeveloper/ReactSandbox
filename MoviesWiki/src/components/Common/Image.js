import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../styles/Image.css';

const Image = ({ image }) => <div className={styles.imagecontainer}>
        <img className={styles.image} src={image}></img>
    </div>;

Image.propTypes = {
  image: PropTypes.string,
};

export default Image;
