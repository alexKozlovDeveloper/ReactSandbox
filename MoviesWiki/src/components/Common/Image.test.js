import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Image from "./Image";

configure({ adapter: new Adapter() });

test('Image rendering test', () => {

    var image = "someImage";

    const component = shallow(<Image image={image} />);
    expect(shallowToJson(component)).toMatchSnapshot();
});

