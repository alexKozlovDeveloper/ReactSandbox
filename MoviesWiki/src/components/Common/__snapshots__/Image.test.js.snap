// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image rendering test 1`] = `
<div
  className="imagecontainer"
>
  <img
    className="image"
    src="someImage"
  />
</div>
`;
